docs/development/local/setup.rst

```diff
@@ -56,31 +56,287 @@ You can then change/set the environment variables in the `.env` file to your nee
+- ``WEBHOOK_SECRET``: HMAC secret for verifying GitHub webhooks. This should be equal to the webhook secret you set in GitHub.
 
-- `POSTGRES_DB`: Database name
-- `POSTGRES_USER`: Database user
-- `POSTGRES_PASSWORD`: Database password
+Postgres Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+`(root .env file)`
+
+- ``POSTGRES_DB``: Database name
+- ``POSTGRES_USER``: Database user
+- ``POSTGRES_PASSWORD``: Database password
 
 Be sure that the database information here matches those in the application server configuration.
 
-**Application Server Configuration** (server .env file):
-
-- `DATASOURCE_URL`: URL to the database
-- `DATASOURCE_USERNAME`: Database username
-- `DATASOURCE_PASSWORD`: Database password
-- `NATS_SERVER`: NATS server URL
-- `NATS_AUTH_TOKEN`: Authorization token for NATS server
-- `REPOSITORY_NAME`: Name of the repository that should be synced
-- `ORGANIZATION_NAME`: Name of the organization/user of the repository that should be synced
-- `GITHUB_AUTH_TOKEN`: GitHub personal access token
-- `RUN_ON_STARTUP_COOLDOWN`: When server starts, it first checks the latest run of sync, if it is less than this value in minutes, it will not run the sync again
-- `OAUTH_ISSUER_URL`: URL to Keycloak realm
+Application Server Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+`(server .env file)`
+
+You can configure **Helios** to work with **either** a Personal Access Token
+(``GITHUB_AUTH_TOKEN``) **or** a GitHub App. **Full functionality** (including
+Helios's custom deployment protection rules) is only available if you configure
+a GitHub App.
+
+**If using a GitHub Personal Access Token**:
+
+- ``GITHUB_AUTH_TOKEN``: Your personal access token (PAT). *(This enables basic GitHub operations but not custom deployment protection rules.)*
+
+**If using a GitHub App** (recommended for full functionality):
+
+- ``GITHUB_APP_NAME``: Name of your GitHub App. After creation, you can confirm the "actual" URL-safe name in the GitHub UI or from the App URL in your browser. For example, if the GitHub UI shows "Helios (AET)" but the URL is ``https://github.com/organizations/ls1intum/settings/apps/helios-aet``, then ``GITHUB_APP_NAME=helios-aet``.
+- ``GITHUB_APP_ID``: Numeric ID of your GitHub App (from GitHub App settings).
+- ``GITHUB_CLIENT_ID``: Client ID (from GitHub App settings).
+- ``GITHUB_PRIVATE_KEY_PATH``: **Absolute** or **relative** path to the **PKCS#8**-formatted private key file associated with your GitHub App. (See note below on converting from PKCS#1 to PKCS#8.)
+- ``ORGANIZATION_NAME``: Name of the GitHub organization/user **only if** you do **not** know the installation ID and want Helios to **auto-detect** it at runtime. Helios will retrieve the GitHub App installations for the given organization/user name and pick the correct installation.
```

Comment:
For me it is not working without setting the organization_name variable. Maybe there is something wrong with the defaults?